### PR TITLE
Ticket/253 mini landing page test

### LINKED ISSUE #253: CMS: Mini Landing Page Test - creation of content
## As a cms user I want to be able to create Mini Landing Page content type to promote Mini Landing Page

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin
Feature: As a cms user I want to be able to create Mini Landing page content type to promote Mini Landing Page

  Scenario: User is adding new Mini Landing page content type with Content Block
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user clicks on "Add content" action button
        And user clicks on "Mini Landing Page" content type
        Then page title is "Mini Landing Page"
        When user selects test site section
        And user fills out the following fields
            | fieldLabel               | value                                                | field_name                     |
            | Pretty URL               | mini-landing-page                                    | field_pretty_url               |
            | Page Title               | Automated Test Mini Landing Page                     | title                          |
            | Browser Title            | Automated Test Mini Landing Page - Browser Title     | field_browser_title            |
            | Card Title               | Automated Test Mini Landing Page - Card Title        | field_card_title               |
            | Meta Description         | Automated Test Mini Landing Page Meta Description    | field_page_description         |
            | Feature Card Description | Automated Test Mini Landing Page - Feature Card Desc | field_feature_card_description |
        And user enters "Mini Landing Page List Description" into "List Description" text field
        And user selects 1 Promotional Image from the list of images
        And user remembers the source of selected promo image for further verification
        And user selects "Add Content Block" from "Contents" dropdown
        And user enters "Content Heading" into "Content Heading" text field
        And user fills out "HTML Content" text area with "This is HTML content for Content Block."
        And user selects "Posted Date" checkbox
        And user selects "Reviewed Date" checkbox
        And user selects "Updated Date" checkbox
        And user selects "Include in search" from "Search Engine Restrictions" dropdown
        When user saves the content page


    Scenario: Add Two Item Feature Card Row section
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user clicks on title with url "mini-landing-page" from the list of content
        And user clicks on the tool bar status green button "Draft"
        And user clicks "View in edit form" button from other actions
        And user selects "Add Two Item Feature Card Row" from "Contents" dropdown
        And user clicks on "Featured Item" link in the "Internal Feature Card" text area
        And user clicks on "Select content" button in the "Internal Feature Card" text area
        And user selects "Article to test Related Resources" item from the list
        And user clicks on "Select content" button in Internal Feature Card area
        And "Article to test Related Resources" had been selected
      And user selects "Add External Feature Card" from Cards dropdown
         And user fills out the following fields
            | fieldLabel               | value                                                | field_name                     |
            | Featured Item Url               | https://www.externalcard.com                                  | field_landing_contents[2][subform][field_two_item_row_cards][4][subform][field_featured_url][0][uri]            |
|Card Title| External Feature Card| field_landing_contents[2][subform][field_two_item_row_cards][4][subform][field_override_card_title][0][value]|
|Card Description| External Feature Card Description|field_landing_contents[2][subform][field_two_item_row_cards][4][subform][field_override_card_description][0][value]|
        When user saves the content page

    Scenario: Add list section
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user clicks on title with url "mini-landing-page" from the list of content
        And user clicks on the tool bar status green button "Draft"
        And user clicks "View in edit form" button from other actions
        And user selects "Add List" from "Contents" dropdown
        And user fills out the following fields
            | fieldLabel               | value                                                | field_name                     |
|Title|List Item Title|field_landing_contents[4][subform][field_list_title][0][value]|
        And user selects "Title, Description, and Image" from "List Item Style" dropdown
        And user clicks on "Link" link in the "Internal Link" text area within List Items
        And user clicks on "Select content" button in the "Internal Link" text area
        And user selects "Article to test Related Resources" item from the list
        And user clicks on "Select content" button to select item
       And browser waits
        And user selects "Add External Link" from "List Item" dropdown
        Then "External Link" section appears
        And user fills out the following fields
            | fieldLabel | value                  | field_name                                                                                       |
            | Link       | https://www.google.com | field_landing_contents[4][subform][field_list_items][1][subform][field_external_link][0][uri]        |  
            | Title      | Google Link            | field_landing_contents[4][subform][field_list_items][1][subform][field_override_title][0][value] |
        And user selects "Add Media Link" from "List Item" dropdown
        Then "Media Link" section appears
        And user clicks on "Link" button to link to a media
        And user clicks on "Select media" to choose a resource to link
        And user selects "Test File for Related Resources" item from the media list
        And user clicks on "Select media" button to select media
        And "Test File for Related Resources" had been selected
        And user fills out the following fields
            | fieldLabel     | value                     | field_name                                                                                       |
            | Override Title |Override Test File for Related Resources| field_landing_contents[4][subform][field_list_items][2][subform][field_override_title][0][value] |
        When user saves the content page

    Scenario: Add Borderless Full-Width Card and raw html
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user clicks on title with url "mini-landing-page" from the list of content
        And user clicks on the tool bar status green button "Draft"
        And user clicks "View in edit form" button from other actions
        And user selects "Add Borderless Full-Width Card" from "Contents" dropdown
        And user clicks on "Featured Item" link in the "Borderless Full-Width Card" text area
        And user clicks on "Select content" button in the "Internal Link" text area
        And user selects "Article to test Related Resources" item from the list
        And user clicks on "Select content" button to select item
        And user selects "Accent warm" from "Card Theme" dropdown
        And user selects "Image Left" from "Image Position" dropdown
        And Button Display dropdown has default value "Show Button"
        And user selects "Show Button" from "Button Display" dropdown
    And user fills out the following fields
            | fieldLabel     | value                     | field_name                                                                                       |
|Button Text|Button text for Borderless Full-Width Card|field_landing_contents[5][subform][field_button_text][0][value]|
|Button Alt Text|Button Alt text for Borderless Full-Width Card|field_landing_contents[5][subform][field_button_alt][0][value]|
        And user selects "Add Raw Html Content" from "Contents" dropdown
        And user enters '<div class="callout-box" tabindex="0"><p>This is to test Raw HTML Content</p></div>'
        When user saves the content page


  Scenario: Add Dynamic List section
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user clicks on title with url "mini-landing-page" from the list of content
        And user clicks on the tool bar status green button "Draft"
        And user clicks "View in edit form" button from other actions
        And user selects "Dynamic List" from "Contents" dropdown
  And user fills out the following fields
            | fieldLabel               | value                                                | field_name                     |
|List Title|Dynamic List Title|field_landing_contents[9][subform][field_list_title][0][value]|
        And user selects "Press Releases" from "Source View" dropdown
        And user selects "Archive Block" from "Display" dropdown
        And user clicks "Options" link in the Dynamic List area
        And user checks "Include View Title" checkbox
        And user enters "5" into "Items per page" dropdown
        And Pagination dropdown has default value "Default settings"
        And user selects "Mini pager" from "Pagination" dropdown
        And user selects "Published" from "Save as" dropdown
        When user saves the content page
    
```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by ahmadakbaryNIH**
closes #253

### CODE REVIEW

**Comment by ahmadakbaryNIH** [comment]
image
image
image

File: cypress.config.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ module.exports = defineConfig({
     admin_username: 'admin',
     admin_password: '',
     test_site_section: '/about-cancer/understanding',
-    front_end_base_url: 'https://www-dev-ac.cancer.gov'
+    front_end_base_url: 'https://www-test-acsf.cancer.gov'
```
Comment (KateMashkinaNIH) [inline cypress.config.js:17]: revert this change - it should not be committed, you only change values to test locally and revert back after

File: cypress.config.js [hunk 2]
```diff
@@ -24,6 +24,6 @@ module.exports = defineConfig({
       return require('./cypress/plugins/index.js')(on, config)
     },
     specPattern: 'cypress/e2e/**/*.feature',
-    baseUrl: 'https://www-dev-ac.cancer.gov',
+    baseUrl: 'https://www-cms-test.cancer.gov',
```
Comment (KateMashkinaNIH) [inline cypress.config.js:27]: same here - revert the change pls

File: cypress/e2e/CMS/$MiniLandingPageTest/index.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        imageSrc = $el[0].getAttribute('src')
+    })
+})
+
+And('user selects {string} content item', (dropDown) => {
+    cy.get(`input[value='${dropDown}']`).click({ force: true })
+})
+
+And('user enters {string} into Content Heading text field', (value) => {
+    cy.get("div[class='ajax-new-content'] input[class='js-text-full text-full form-text']").type(value)
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$MiniLandingPageTest/index.js:23]: bad locator - use input[name^='field_landing_contents[0][subform][field_content_heading]'] instead

File: cypress/e2e/CMS/$MiniLandingPageTest/index.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+And('user clicks on {string} button in the Internal Feature Card text area', (title) => {
+    cy.get(`input[value='${title}']`).click({ force: true })
+})
+
+And('user clicks on {string} button to select item', (title) => {
+    cy.getIframeBody('iframe[name="entity_browser_iframe_cgov_content_browser"]').find(`input[value="${title}"]`).click({ force: true })
+})
+
+And('user clicks on {string} button in Internal Feature Card area', (dropDown) => {
+    cy.get(`input[value='${dropDown}']`).click({ force: true })
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$MiniLandingPageTest/index.js:43]: this step doesn't exist in the feature file - remove

File: cypress/e2e/CMS/$MiniLandingPageTest/index.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+And('user selects {string} checkbox', (visible) => {
+    cy.get(`[class='fieldset-wrapper']:contains("${visible}")`).parent().find('input.form-checkbox').check({ force: true })
+})
+
+And('user clicks on {string} link in the {string} text area', (title, cartOption) => {
+    cy.get(`div.paragraph-type-title:contains('${cartOption}')`).parent().parent().find(`span:contains('${title}')`).parent().click()
+})
+
+And('user clicks on {string} button in the Internal Feature Card text area', (title) => {
+    cy.get(`input[value='${title}']`).click({ force: true })
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$MiniLandingPageTest/index.js:35]: this step doesn't exist in the feature file - remove

File: cypress/e2e/CMS/$MiniLandingPageTest/index.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+And('user selects {string} from Display dropdown',(dropdown) => {
+    cy.get("select[name='field_landing_contents[4][subform][field_source_view][0][display_id]']").select(dropdown)
+})
+
+
+
+
+
+
+
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$MiniLandingPageTest/index.js:127]: remove all of the extra empty lines here

**Review by KateMashkinaNIH — CHANGES_REQUESTED** [review]

**Review by KateMashkinaNIH — APPROVED** [review]

**Pull request merged**